backend/src/StdLibExecution/Libs/Json.fs

```diff
@@ -410,16 +411,20 @@ let parse
 
       | CustomType.Record (firstField, additionalFields) ->
         let fieldDefs = firstField :: additionalFields
+        let enumerated = j.EnumerateObject() |> Seq.toList
+
         let dvalMap =
-          j.EnumerateObject()
-          |> Seq.map (fun jp ->
-            let correspondingType =
-              fieldDefs
-              // TODO: handle case where field isn't found
-              |> List.find (fun def -> def.name = jp.Name)
-              |> fun def -> def.typ
-            let converted = convert correspondingType jp.Value
-            (jp.Name, converted))
+          fieldDefs
+          |> List.map (fun def ->
+            let correspondingValue =
+              enumerated
+              // TODO: handle case where value isn't found for
+              // and maybe, if it's an Option<>al thing, don't complain
+              |> List.find (fun v -> v.Name = def.name)
+              |> fun field -> field.Value
```

Comment:
WDYT should happen for:
```fsharp
type Person = { Age: Option<Int>; Name: String }
JSON.parse<Person> """{ "Name": "John" }"""
```
?

I'm inclined to think the Age should just be set to None.